src/main/java/hudson/plugins/sonar/SonarGlobalConfiguration.java

```diff
@@ -145,36 +155,27 @@ public static SonarGlobalConfiguration get() {
 
   @SuppressWarnings("unused")
   public ListBoxModel doFillCredentialsIdItems(@QueryParameter String credentialsId) {
-    if (!Jenkins.getInstance().hasPermission(Jenkins.ADMINISTER)) {
-      return new StandardListBoxModel().includeCurrentValue(credentialsId);
-    }
-
-    return new StandardListBoxModel()
-      .includeEmptyValue()
-      .includeMatchingAs(
-        ACL.SYSTEM,
-        Jenkins.getInstance(),
-        StringCredentials.class,
-        Collections.emptyList(),
-        CredentialsMatchers.always()
-      );
+    return constructListBoxModel(credentialsId);
```

Comment:
Nitpick: I think "create" is just a more common word for the pattern than "construct".

And definitely good to not duplicate so much code! (I should have caught earlier.)